federation/executor.go

```diff
@@ -105,28 +105,59 @@ func (e *Executor) runOnService(ctx context.Context, service string, typName str
 
 	// If it is not a root query, nest the subquery on the federation field
 	// and pass the keys in to find the object that the subquery is nested on
+	// Pass all federated keys for that service as arguments 
 	// {
 	//    __federation {
-	//     [ObjectName] (keys: Keys) {
+	//     [ObjectName]-[Service] (keys: Keys) {
 	//       subQuery
 	// 		}
 	//   }
 	// }
 	isRoot := keys == nil
 	if !isRoot {
+		federatedName := fmt.Sprintf("%s-%s", typName, service)
+		newKeys := make(map[string]interface{}, len(keys))
+
+		var rootObject *graphql.Object
+		var ok bool
+		for f, _ := range e.planner.schema.Fields {
+			if f.Type.String() == typName {
+				rootObject, ok = f.Type.(*graphql.Object)
+				if !ok {
+					return nil, oops.Errorf("root object isn't a graphql object")
+				}
+			}
+		}
+
+		// If it is a federated key on that service, add it to the input args
+		// passed in to the federated field func as one of the federatd keys 
+		for name, key := range keys[0].(map[string]interface{}) {
```

Comment:
why only use keys[0]?

Reply:
If so, can we change the key to be an inferface{} instead of a slice of interface{}?

Reply:
I misspoke earlier, the federated field returns a list of objects, so we actually want to parse all the keys. I updated this along with fixing the executor to support this.